eZ/Publish/Core/MVC/Symfony/Locale/UserLanguagePreferenceProvider.php

```diff
@@ -59,7 +74,7 @@ public function getPreferredLanguages(): array
             $locale = strtolower($locale);
             if (isset($this->languageCodesMap[$locale])) {
                 $languageCodes = array_merge($languageCodes, $this->languageCodesMap[$locale]);
-            } elseif (preg_match('/^([a-z]{3})-([a-z]{2})$/', $locale, $matches)) {
+            } elseif (preg_match('/^([a-z]{3})_([a-z]{2})$/', $locale, $matches)) {
```

Comment:
Given this will never occur in normal environment, but just covers test case I've added some time ago, I think we should drop it and drop `pol-PL` in specific data providers.
```php
        foreach ($this->getPreferredLocales() as $locale) {
            $locale = strtolower($locale);
            if (!isset($this->languageCodesMap[$locale])) {
                continue;
            }

            $languageCodes = array_merge($languageCodes, $this->languageCodesMap[$locale]);
        }
```